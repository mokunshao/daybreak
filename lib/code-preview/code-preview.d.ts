import React, { HtmlHTMLAttributes } from 'react';
import './code-preview.scss';
interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
    code: string;
}
export declare const CodePreview: React.FunctionComponent<Props>;
export default CodePreview;
//# sourceMappingURL=code-preview.d.ts.map