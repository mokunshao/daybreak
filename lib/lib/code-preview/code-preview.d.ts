import React, { HtmlHTMLAttributes } from 'react';
import './code-preview.scss';
interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
    code: string;
}
declare const CodePreview: React.FunctionComponent<Props>;
export default CodePreview;
//# sourceMappingURL=code-preview.d.ts.map