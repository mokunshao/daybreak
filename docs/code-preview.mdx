---
title: CodePreview
---

import {useState} from 'react'
import { Playground, Props } from 'docz'
import CodePreview from '../src/code-preview/code-preview.tsx'

# CodePreview

```javascript
import { CodePreview } from 'daybreak';
```

## Basic

`CodePreview` has no code highlighting, just like `Stack Overflow`.

`CodePreview` 没有代码高亮，就像 `Stack Overflow` 一样。

<Playground>
  {()=>{
    const code = `
public class Main {
    public static void main(String[] args) {
    System.out.println("Hello, World!");
  }
}
    `;
    return (
      <div>
        <CodePreview code={code.trim()} />
      </div>
    );
  }}
</Playground>

## Props

<Props of={CodePreview} />
