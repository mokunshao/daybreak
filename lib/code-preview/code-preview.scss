.daybreak-code-preview {
  &-button-wrapper {
    padding: 0.5em 0;
  }

  &-content {
    background: #eee;
    padding: 1em;
    display: inline-block;
  }
}
