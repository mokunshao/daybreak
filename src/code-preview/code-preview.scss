.daybreak-code-preview {
  &-button-wrapper {
    padding: 0.3em 0;
  }

  &-content {
    background: #eee;
    padding: 0.3em;
    display: inline-block;
    margin: 0;
  }
}
