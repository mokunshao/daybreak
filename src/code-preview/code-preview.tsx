import React, { useState, HtmlHTMLAttributes } from 'react';
import { Button } from '../button/button';
import { classes } from '../utils/classes';
import { joinedClass } from '../utils/joinedClass';
import './code-preview.scss';

const codePreview = joinedClass('code-preview');

interface Props extends HtmlHTMLAttributes<HTMLDivElement> {
  code: string;
}

export const CodePreview: React.FunctionComponent<Props> = (props) => {
  const [codeVisible, setCodeVisible] = useState(false);
  const { code, className } = props;
  const preview = <pre className={codePreview('content')}><code>{code}</code></pre>;
  return (
    <div className={classes(codePreview(), className)}>
      <div className={codePreview('button-wrapper')}>
        <Button type="button" onClick={() => setCodeVisible(!codeVisible)}>Code Preview</Button>
      </div>
      {codeVisible && preview}
    </div>
  );
};

export default CodePreview;
