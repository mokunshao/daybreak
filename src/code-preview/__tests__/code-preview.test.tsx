import React from 'react';
import { render } from '@testing-library/react';
import { CodePreview } from '../code-preview';

function Example() {
  const code = `
public class Main {
    public static void main(String[] args) {
    System.out.println("Hello, World!");
  }
}
    `;
  return (
    <div>
      <CodePreview code={code.trim()} />
    </div>
  );
}

describe('CodePreview', () => {
  it('exits', () => {
    expect(CodePreview).toBeTruthy();
    expect(<Example />).toBeTruthy();
  });

  it('match snapshot', () => {
    const { asFragment } = render(<Example />);
    expect(asFragment()).toMatchSnapshot();
  });
});
