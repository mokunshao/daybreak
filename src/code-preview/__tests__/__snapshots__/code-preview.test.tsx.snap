// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePreview match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="daybreak-code-preview"
    >
      <div
        class="daybreak-code-preview-button-wrapper"
      >
        <button
          class="daybreak-button daybreak-button-normal"
          type="button"
        >
          Code Preview
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
